[lldb] Correct bridgeOS -> BridgeOS spelling

JDevlieghere opened the pull request:
The BridgeOS SDK is capitalized, but previously failed to parse because
we were looking for bridgeOS. This PR updates the enum value and the
canonical spelling.

rdar://162641896

adrian-prantl commented on lldb/source/Utility/XcodeSDK.cpp:
```diff
@@ -65,27 +65,27 @@ bool XcodeSDK::operator==(const XcodeSDK &other) const {
 
 static XcodeSDK::Type ParseSDKName(llvm::StringRef &name) {
-  if (name.consume_front("MacOSX"))
+  if (name.consume_front_insensitive("MacOSX"))
```
I think I would prefer us remaining strict and just fix the spelling of bridgeOS going forward.

JDevlieghere committed:
[lldb] Correct bridgeOS -> BridgeOS spelling

The BridgeOS SDK is capitalized, but previously failed to parse because
we were looking for bridgeOS. This PR updates the enum value and the
canonical spelling.

rdar://162641896
